File: projects/03/clojure/src/letter_grade.clj
;;;; letter_grade.clj is a driver for function yearCode.
;;;;
;;;; Input: A number grade
;;;; Output: The corresponding letter grade
;;;;
;;;; Usage: clojure -m year_codes
;;;;
;;;; Begun by: Prof. Adams, for CS 214 at Calvin College.
;;;; Completed by: Sarah Whitten
;;;; Date: February 19, 2019
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns letter_grade)                    ; name the program

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; letterGrade() returns the letter for a given grade.
;;; Receive: grade, a string.
;;; Return: the corresponding letter grade
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(defn parse-int [s]
   (Integer. (re-find  #"\d+" s ))
)

(defn letterGrade [grade]
    (cond
        (= 10 (/ (parse-int grade) 10))       "A"
        (<= 9 (/ (parse-int grade) 10))  "A"
        (<= 8 (/ (parse-int grade) 10))  "B"
        (<= 7 (/ (parse-int grade) 10))  "C"
        (<= 6 (/ (parse-int grade) 10))  "D"
        :else                                 "F"
    )
)

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; Function -main() test-drives our yearCode() function.
;;; Input: year: one of {freshman, sophomore, junior, or senior}.
;;; Output: the code corresponding to year.
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defn -main []
 (print "\nEnter your grade: ") (flush)
 (let
    [ grade (read-line) ]
    (println (letterGrade grade))      ; display its code
 )
)

File: projects/03/java/LetterGrade.java
/* LetterGrade.java is a driver for function letterGrade().
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 ****************************************************************/

import java.util.Scanner;

public class LetterGrade {

    public static void main(String[] args) {
      Scanner keyboard = new Scanner(System.in);
      System.out.println("\nEnter your number grade: "); // prompt

      String grade_str = keyboard.next();       // read grade
      System.out.println(letterGrade(grade_str)); // display its letter
    }


   /***************************************************************
    * letterGrade() converts an number year into its letter equivalent.
    *
    * Receive: grade_str, a string.
    * Return: the letter grade corresponding to number grade.
    */

    public static String letterGrade(String grade_str) 
    {
        int grade = Integer.parseInt(grade_str);
        switch (grade / 10) {
            case 10: case 9: return "A";
            case 8: return "B";
            case 7: return "C";
            case 6: return "D";
            default: return "F";
        }
    }
}